### DISCUSSION TITLE
[나래] assignment 기능구현

### DISCUSSION THREAD

**Discussion by NR0617** [Wanted-07-team-9/wanted_assignment_02#20] [Ideas]
## Assignment 1 이슈 목록 및 상세 화면 기능 구현

- 이슈 목록 4개에 광고가 들어가도록 구현
```js
              {issueDataArr.map((el, idx) => {
                if (idx < 4) {
                  return (<IssueList />);
                } else if (idx === 4) {
                  return (
                    <>
                      <AdsBox />
                      <IssueList
                        key={idx}
                        number={el.number}
                        title={el.title}
                        user={el.user}
                        created_at={el.created_at}
                        comments={el.comments}
                      />
                    </>
              }
```
- react-markdown으로 상세화면 구현
```js
import ReactMarkdown from 'react-markdown';

const DetailPageComponent = ({ detailPageData }) => {
  const markdown = detailPageData.body;

  return ()
}
```

## Assignment 2 Context API를 활용한 API 연동 
- main 페이지에서 context API로 데이터를 불러오도록 제작
```js
export const getDataList = async (dispatch, page) => {
  dispatch({ type: 'GET_DATALIST' });
  try {
    const response = await axios.get(
      `https://api.github.com/repos/angular/angular-cli/issues?sort=comments&state=open&per_page=7&page=${page}`,
      {
        headers: {
          'Content-Type': 'application/json',
          Authorization: `${process.env.REACT_APP_GITHUB_TOKEN}`,
        },
      }
    );
    dispatch({ type: 'GET_DATALIST_SUCCESS', data: response.data });
  } catch (e) {
    dispatch({ type: 'GET_DATALIST_ERROR', error: e });
  }
};

```

## Assignment 3 데이터 요청 중 로딩 표시
- 데이터 로딩중 로딩컴포넌트가 보이도록 구현

```js
      {isLoading && <Loading />}
```

## Assignment3 에러 화면 구현

```js
//App.js
 <Route path="*" element={<ErrorPage />} />

//ErrorPage.js
const ErrorPage = () => {
  return <p>잘못된 요청입니다</p>;
};

export default ErrorPage;

```
## Assignment 4 지정된 조건(open 상태, 코멘트 많은 순)에 맞게 데이터 요청 및 표시
- 데이터를 요청할 때 엔드포인트에 ?sort=comments&state=open 를 추가하여 조건에 맞게 데이터를 요청 및 표시
```js
export const getIssueList = async () => {
  const issueData = await get({
    url: '/angular/angular-cli/issues',
    data: 'sort=comments&state=open&per_page=7',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `${process.env.REACT_APP_GITHUB_TOKEN}`,
    },
  });

  return issueData.data;
};
```

## Assignment 5 반응형 웹 구현(UI는 데스크톱, 모바일에서 보았을 때 모두 읽기 편하게 구현) 
- 768px 이하에서 잘 보일 수 있도록 미디어 쿼리를 추가
```js
import styled from 'styled-components';

export const PageContainer = styled.div`
  padding: 10em;
  @media screen and (max-width: 768px) {
    padding: 8em 0 0 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
`;
```
1. 이슈 목록 화면
    - [X]  이슈 목록 가져오기 API 활용
    - [X]  open 상태의 이슈 중 코멘트가 많은 순으로 정렬
    - [X]  각 행에는 ‘이슈번호, 이슈제목, 작성자, 작성일, 코멘트수’를 표시
    - [X]  다섯번째 셀에는 광고 이미지 출력
        - [X]  이미지 사용
            
            [https://image.wanted.co.kr/optimize?src=https%3A%2F%2Fstatic.wanted.co.kr%2Fimages%2Fuserweb%2Flogo_wanted_black.png&w=110&q=100](https://image.wanted.co.kr/optimize?src=https%3A%2F%2Fstatic.wanted.co.kr%2Fimages%2Fuserweb%2Flogo_wanted_black.png&w=110&q=100)
            
            [https://image.wanted.co.kr/optimize?src=https%3A%2F%2Fstatic.wanted.co.kr%2Fimages%2Fuserweb%2Flogo_wanted_black.png&w=110&q=100](https://image.wanted.co.kr/optimize?src=https%3A%2F%2Fstatic.wanted.co.kr%2Fimages%2Fuserweb%2Flogo_wanted_black.png&w=110&q=100)
            
        - [X]  광고 이미지 클릭 시 [https://www.wanted.co.kr/](https://www.wanted.co.kr/) 로 이동
    - [X]  화면을 아래로 스크롤 할 시 이슈 목록 추가 로딩(인피니티 스크롤)

1. 이슈 상세 화면
    - [X]  이슈의 상세 내용 표시
    - [X] ‘이슈번호, 이슈제목, 작성자, 작성일, 코멘트 수, 작성자 프로필 이미지, 본문' 표시
    
2. 공통 헤더
    - [X]  두 페이지는 공통 헤더를 공유합니다.
    - [X]  헤더에는 Organization Name / Repository Name이 표시됩니다.
    
- 필수 요구 사항
    - [X] 이슈 목록 및 상세 화면 기능 구현
    - [X] Context API를 활용한 API 연동
    - [X] 데이터 요청 중 로딩 표시
    - [X] 에러 화면 구현
    - [X] 지정된 조건(open 상태, 코멘트 많은 순)에 맞게 데이터 요청 및 표시
    - [X] 반응형 웹 구현(UI는 데스크톱, 모바일에서 보았을 때 모두 읽기 편하게 구현)

**Comment by songseul** [comment]
# Assignment 4 지정된 조건(open 상태, 코멘트 많은 순)에 맞게 데이터 요청 및 표시
- 토큰 값을 env 파일에 작성하고 사용 하셔서 노출성이 줄어들어 보안성이 높아진것 같습니다.
